lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -160,6 +161,10 @@ def squeeze_priorities
     MiqAeDomain.reset_priority_by_ordered_ids(ids)
   end
 
+  def destroy_repository
+    git_repository.destroy if git_repository
+  end
+
```

Comment:
Thoughts on removing the `before_destroy` and adding a dependency to the belongs_to cause?:
```
belongs_to :git_repository, :dependent => :destroy
```

Whenever you do an `miq_ae_domain.destroy`, it should then go through the `git_repository.destroy` routine as well (all the callbacks, etc).